### ISSUE TITLE
[Linux] Hidden files synced only on startup of owncloud client

### ISSUE THREAD

**Issue by tmysik** [owncloud/client#3980]
With v2.0, owncloud client stopped syncing of hidden files. I enabled it in its settings again, however these hidden files are synced only right after the startup of owncloud client. In other words, if I change a hidden file manually, no upload is done. I have to quit owncloud client and start it again, then the changed hidden file is correctly synced.

I am on Kubuntu 15.04, ownloud v2.0.1.

Thanks.

**Comment by guruz** [comment]
Maybe somehow related to the file watcher not getting the updated setting?

**Comment by dragotin** [comment]
@tmysik we're about to release 2.0.2 and I'd ask you to verify with that. There were changes in regards to hidden files and the filesystem watcher.

**Comment by tmysik** [comment]
@dragotin I just updated to v2.0.2 but unfortunately, there is no difference in syncing of hidden files. To be precise, I sync directory ```.private``` which contains some files and other directories (with other files, of course).

At least, sync still works fine if I manually restart ownCloud client :)

Thanks.

**Comment by dragotin** [comment]
I can confirm that it does not work with 2.0.2. However, it works with current master, which will become the next release called 2.1.

**Comment by tmysik** [comment]
@dragotin 

> I can confirm that it does not work with 2.0.2. However, it works with current master, which will become the next release called 2.1.

Great to hear! Thanks!

**Comment by dragotin** [comment]
Well, it would be good to understand what has changed this behaviour.... @ckamm do you have an idea?

**Comment by dragotin** [comment]
@Dianafg76 can you please add a testcase to your testplans?

**Comment by Dianafg76** [comment]
of course

**Comment by tmysik** [comment]
@dragotin 

> However, it works with current master, which will become the next release called 2.1.

Hi all, can anyone please verify that it works for version 2.1.0? I just installed it (from official OpenSuse repository) and it still does not work for me :/ Fortunately, the trick with manual client restarting still works.

Thanks a lot.

**Comment by guruz** [comment]
@tmysik How many folders do you have in your sync dir? A lot? Then would you please test what was described in https://github.com/owncloud/client/issues/4107 ?

**Comment by tmysik** [comment]
@guruz 

> How many folders do you have in your sync dir? A lot?

I do not think so:

```
gapon@cattie ~/storage/ownCloud $ find . -type d | wc -l
196
gapon@cattie ~/storage/ownCloud $ cat /proc/sys/fs/inotify/max_user_watches
524288
```

Moreover, as I wrote, manual restart of ownCloud client solves the problem for me - that would not help, right? Or am I wrong?

Thanks!

**Comment by guruz** [comment]
@tmysik I also wonder if https://github.com/owncloud/client/issues/3475 is related. Is your sync folder a symlink?

But still why would it only be for hidden files..

**Comment by guruz** [comment]
@tmysik Can you try this:
Go to "Edit ignored files", click the "Sync hidden files" once, Press "OK". Go again to  "Edit ignored files", click the "Sync hidden files" once, Press "OK". (so that it is checked at your last step).
Does it work then?

**Comment by tmysik** [comment]
@guruz 

> Is your sync folder a symlink?

No symlinks in my sync folder:

```
gapon@cattie ~/storage/ownCloud $ find . -type l | wc -l
0
```

**Comment by tmysik** [comment]
@guruz 
> Does it work then?

Did not seem to help.

The strange thing is that sometimes the sync happens but a bit later (no idea why and when).

Guys if I am the only one who sees this problem then it is apparently something with my PC/setup and it makes no sense to waste your time on it. Or is there anyone else who is able to reproduce this issue?

Thanks.

**Comment by guruz** [comment]
(I couldn't on OS X... someone else shall try then)

**Comment by ckamm** [comment]
@tmysik I also can't reproduce the problem on linux.

**Comment by tmysik** [comment]
@ckamm Thanks for trying it. In such a case, feel free to close this issue. However, as I wrote in my initial comment, it worked for me without any problem with ownCloud client v1.x.

Thanks.

**Comment by ckamm** [comment]
@tmysik I'm sorry we can't help! If you have any extra information or figure it out, please let us know!

**Comment by tmysik** [comment]
@ckamm Sure, will do that. Thank you!

**Comment by tmysik** [comment]
BTW this issue has been solved with version 2.2.0 - so, thanks a lot!

**Comment by guruz** [comment]
👍

**Comment by PVahr** [comment]
Had same problem while trying to sync a .git folder.
Running Mac OS Catalina 10.15, owncloud desktop 2.9.2.

After selecting "sync hidden files", they were not synchronized at all.
Restarting the desktop application did the job.

[I actually did not want to sync all the hidden files, but just the .git one. Wouldn't it be better to have some exclusion pattern tool? Like on git itself, a command to say "do not sync any hidden files except ..." ?]

**Comment by TheOneRing** [comment]
The need to restart the client was fixed in 2.10
